apis/v1beta1/validation/httproute.go

```diff
@@ -287,3 +289,44 @@ func hasExactlyOnePrefixMatch(matches []gatewayv1b1.HTTPRouteMatch) bool {
 
 	return true
 }
+
+// ValidateHTTPParentRefs validates that if ParentRefs includes 2 or more references
+// to the same parent (based on kind, name, and namespace), those ParentRefs must
+// specify unique SectionName values.
+func ValidateHTTPParentRefs(parentRefs []gatewayv1b1.ParentReference, path *field.Path) field.ErrorList {
```

Comment:
So it seems to me that while we name this with `HTTP` we could just as easily apply this across other APIs. Is it worth making this validation more generic and plugging it in to validation for all routes kinds?

Reply:
Yes, I agree. I will create a generic validation than can be used for all routes kinds.